cdc/log.cc

```diff
@@ -663,6 +663,109 @@ auto make_maybe_back_inserter(Container& c, const abstract_type& type, collectio
     return maybe_back_insert_iterator<Container, T>(c, type, s);
 }
 
+static size_t collection_size(const bytes_opt& bo) {
+    if (bo) {
+        bytes_view bv(*bo);
+        return read_collection_size(bv, cql_serialization_format::internal());
+    }
+    return 0;
+}
+template<typename Func>
+static void udt_for_each(const bytes_opt& bo, Func&& f) {
+    if (bo) {
+        bytes_view bv(*bo);
+        std::for_each(tuple_deserializing_iterator::start(bv), tuple_deserializing_iterator::finish(bv), std::forward<Func>(f));
+    }
+}
+static bytes merge(const collection_type_impl& ctype, const bytes_opt& prev, const bytes_opt& next, const bytes_opt& deleted) {
+    std::vector<std::pair<bytes_view, bytes_view>> res;
```

Comment:
Could you not have done this so git registered it as a change instead of a replace? Seems we loose a lot of traceability here. Perhaps not the most important, but it would perhaps be nice to be able to relate what changes are actual changes and which are just moving code about...

Reply:
Well this does have a completely separate commit literally titled "move some functions", so is it such a big problem?

Reply:
I don't know if it would, but I could split it further: first move without changing indentation, then change indentation.

Reply:
It is perhaps not these exact functions - they are after all fairly simple, and any lack of explanation in them is my fault. I am just a bit concerned about replacing all code in a fashion where we pretty much cannot git blame beyond this to trace why a particular code line exists (I mean the reasoning behind it). 
Just to be clear, I don't consider this a showstopper, and I am generally for the changes. Just a bit worried.

Reply:
Well, replacing code is the whole purpose of refactoring, isn't it? :P
I don't see how things can be improved with regards to traceability.

As for the question why a particular line of code exists: I don't think it was easy to answer previously either.
IMO the only thing we can do is write good comments. Which I'll try to do next.

If someone wants to look how the code previously looked, they can look into the commit history. I often find in Scylla that "git blame" doesn't work because there were code movements, and I had to "git log -p", see where the move happened, then either "git blame <commit-hash>" or further "git log -p" (perhaps on a different file).

Reply:
Which is why I have become slightly wary of refactoring... :-) (Rock, glass house!)
Point is just that it is nice when one can retain a sliver of the past. For example in this case, you could pretty much have left the functions as transformer statics.